hollow/src/main/java/com/netflix/hollow/api/client/HollowDataHolder.java

```diff
@@ -177,7 +181,21 @@ private void applyStateEngineTransition(HollowBlobInput in, OptionalBlobPartInpu
             reader.applyDelta(in, optionalPartIn);
         }
 
-        setVersion(transition.getToVersion());
+        long expectedToVersion = transition.getToVersion();
+        Long actualToVersion = null;
+        String actualToVersionStr = stateEngine.getHeaderTag(HEADER_TAG_PRODUCER_TO_VERSION);
+        if (actualToVersionStr != null) {
```

Comment:
nit: can collapse the two if blocks that are checking the same condition

Reply:
not apply anymore